conformance/lang/expressions/unary-expression/unary-minus-expression.balt

```diff
@@ -0,0 +1,893 @@
+    io:println(-b); // @output -2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output -2147483647
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative optional int:Signed32, no overflow scenarios.
+Labels: DecimalNumber, int:Signed32, nil-literal, numeric-nil-lifting, optional-type, unary-minus, value:toBalString
+
+public function main() {
+    int:Signed32? a = -1;
+    int:Signed32? b = -2;
+    int:Signed32? c = ();
+    int:Signed32? d = -2147483648;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output 2147483648
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional user defined sub types of int,
+              no overflow scenarios.
+Labels: DecimalNumber, module-type-defn, numeric-nil-lifting, optional-type, union-type, unary-minus, value:toBalString
+
+
+type Ints -2|-1|0|1|2;
+
+public function main() {
+    Ints? a = 1;
+    Ints? b = ();
+    Ints? c = 0;
+
+    io:println(-a); // @output -1
+    io:println((-b).toBalString()); // @output ()
+    io:println(-c); // @output 0
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative optional user defined sub types of int,
+              no overflow scenarios.
+Labels: DecimalNumber, module-type-defn, numeric-nil-lifting, optional-type, union-type, unary-minus, value:toBalString
+
+type Ints -2|-1|0|1|2;
+
+public function main() {
+    Ints? a = -1;
+    Ints? b = ();
+    Ints? c = -2;
+
+    io:println(-a); // @output 1
+    io:println((-b).toBalString()); // @output ()
+    io:println(-c); // @output 2
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive floats, no overflow scenarios.
+Label: float, DecimalNumber, unary-minus
+
+public function main() {
+    float a = 1.0;
+    float b = 101.5;
+    float c = 0;
+    float d = 922439289.0;
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -101.5
+    io:println(-c); // @output -0.0
+    io:println(-d); // @output -9.22439289E8
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative floats, no overflow scenarios.
+Label: float, DecimalNumber, unary-minus
+
+public function main() {
+    float a = -1.0;
+    float b = -101.5;
+    float c = -922439289.0;
+    io:println(-a); // @output 1.0
+    io:println(-b); // @output 101.5
+    io:println(-c); // @output 9.22439289E8
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive user defined sub types of float,
+              no overflow scenarios.
+Labels: DecimalNumber, module-type-defn, unary-minus, union-type
+
+type Floats -2.0|-1.0|0|1.0|2.0;
+
+public function main() {
+    Floats a = 1.0;
+    Floats b = 2.0;
+    Floats c = 0;
+
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -2.0
+    io:println(-c); // @output -0.0
+}
+
```

Comment:
Let's also add some more tests with subtypes, which seems to be not working with floats too.
```ballerina
const float A = -10f;
const float B = -20f;

type Floats1 -2.0|-1.0|0|1.0|2.0;

type Floats2 A|B;

type Floats3 Floats1|Floats2;

public function main() {
    Floats2 a = -10f;
    Floats3 b = -1.0;

    io:println(-a); // operator not defined
    io:println(-b); // operator not defined
}
```
Let's add tests anyway, and mention the `Fail-Issue`.

Reply:
updated in 6862c07